### ISSUE TITLE
Unable to build git master on Linux (OpenSSL binding)

### ISSUE THREAD

**Issue by johnthagen** [sfackler/rust-native-tls#93]
On a clean Ubuntu 16.04.4 32-bit (the problem seems to happen on 64-bit Linux as well) VM I installed:

```bash
$ sudo apt install gcc pkg-config libssl-dev
```

Project:

```toml
# Cargo.toml
...
[dependencies]
native-tls = { git = "https://github.com/sfackler/rust-native-tls" }
```

`main.rs` is populated with the example client from the `native-tls` README.

The build cannot find `SslOptions::NO_SSL_V2` (Did you mean `NO_SSLV2`) or `SslOptions:NO_TLSV1_0` (Did you mean `NO_TLSV1_1`)?

OpenSSL version:

```bash
$ openssl version
OpenSSL 1.0.2g  1 Mar 2016
```

Confirmed the problem exists on stable Rust 1.26.0 and nightly 2018-05-25.

**Comment by johnthagen** [comment]
If this really is a problem, it's surprisingly that [CircleCI](https://circleci.com/gh/sfackler/rust-native-tls) didn't catch it.

**Comment by johnthagen** [comment]
I think the commit that broke this is: https://github.com/sfackler/rust-native-tls/commit/3436f9642e4640389f4a6d1282303b302328c379#diff-c67038c29bc9227b0ac84a64061cbd31

**Comment by sfackler** [comment]
Oh oops!

CircleCI didn't catch it since the build image is using OpenSSL 1.1.0 which doesn't use that bit of code.

**Comment by johnthagen** [comment]
Confirmed fixed on Ubuntu 16.04.4 OpenSSL 1.0.2g.
